### DISCUSSION TITLE
请教如何让MPV在用鼠标快速拖动进度条的时候指哪儿打哪儿

### DISCUSSION THREAD

**Discussion by Anikatball** [hooke007/mpv_PlayKit#432] [mpv 通用]
使用potplayer或者VLC的时候，用鼠标来回快速拖动进度条的时候可以瞬间响应，非常流畅，但是用mpv-lazy的时候，特别是高码率或者高分辨率视频，快速拖动进度条的时候总会卡顿半秒到一秒左右，不知道有没有让MPV也指哪儿打哪儿的方式，望指教，谢谢。

**Comment by hooke007** [comment]
我不一点都不卡。

（前提是你没用什么特殊东西，否则）
要么是你的瞬时性能不够缩略图，要么是不够auto-copy解码

**Comment by hooke007** [comment]
另外你说的vlc，pot快不会是只跳关键帧吧，我似乎印象他们默认中是这样的。如果是这样的话，uosc默认写死拖动逻辑的。如果你有这个需求，建议换界面吧（比如原始osc默认就是拖动解码关键帧

**Reply by hooke007** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/41094733/9186b50b-bcbc-4bb8-8f12-1dfe60550785)
后边看过了，pot的确默认拖动是只seek关键帧的，这当然比uosc拖动找精确帧快

**Reply by JamYiz** [reply]
> 另外你说的vlc，pot快不会是只跳关键帧吧，我似乎印象他们默认中是这样的。如果是这样的话，uosc默认写死拖动逻辑的。如果你有这个需求，建议换界面吧（比如原始osc默认就是拖动解码关键帧

如果是4K的视频文件，我在进度条上点击跳转PotPlayer会比mpv更快，大佬知道怎么把mpv速度提快一点吗？会卡0.5秒左右

**Reply by hooke007** [reply]
pot不管点击还是拖动都是只seek关键帧。
osc uosc 点击都是只seek精确帧

**Reply by JamYiz** [reply]
> pot不管点击还是拖动都是只seek关键帧。 osc uosc 点击都是只seek精确帧

有办法让mpv也实现这个功能吗？需要该哪些设置啊大佬

**Reply by hooke007** [reply]
有更好的要实现的理由吗。
在我看来“指哪儿打哪儿”比较重要，关键帧的间隔可能间距5秒以上，我不想要点半天点不到我想要的位置。
而且为什么要一直拖来拖去呢，是不想用缩略图吗

另外原始osc就是拖动跳关键帧(我前面已经给出建议了“建议换界面”)

**Reply by verygoodlee** [reply]
uosc没给配置项，要改代码，最简单的改法，直接在这里的116行下面加一行`fast = true`，无视传过来的参数，强制关键帧跳转
https://github.com/hooke007/MPV_lazy/blob/087e6f837df47c8386313acfce8745cb4be8a9ac/portable_config/scripts/uosc/elements/Timeline.lua#L115-L120

**Reply by Anikatball** [reply]
我最近仔细对比了一下，即使是原版mpv的原始osc，4K高码率视频在快速拖动进度条的时候仍然是慢原版potplayer一拍，继续用你这里提供的教程弄了一套pot+lav+madvr（感谢），快速拖动进度条依然是很顺滑。我是5600G的核显，其它的显卡没有试过。你上面提到跳转的精确性比较重要我是同意的，奈何我确实有经常暴力拖动进度条的需求。目前potplayer也有一个短板比较难受，启动速度比mpv慢很多，同样是打开一个视频，mpv基本是瞬间启动并打开，pot启动就慢不少，打开还要转两下，不知道是不是我硬件配置问题.........

**Reply by hooke007** [reply]
osc的渲染和光标的位置不是同步的。而且osc还受到视频帧率的同步限制。
当然也不排除解码的时候用了大量加速技巧跳过大量帧请求。

**Reply by verygoodlee** [reply]
pot启动速度就是慢一大截，跟配置关联性不大。

mpv默认状态下，硬盘速度也会影响seek，放固态盘比机械盘快, 
内存够大的话, 倒是有个可提升seek速度的点，
给大点缓存，启动后全加载到内存里，seek的时候就少了读盘的步骤，速度应该会有所提升。
```ini

cache=yes   # 开启缓存，默认值auto只会在播放网络链接时开启
demuxer-max-bytes=4096MiB       # 向后缓存大小 
demuxer-max-back-bytes=1024MiB  # 前面保留的缓存大小
```

**Reply by hooke007** [reply]
pot和vlc据我观察和mpv的osc最不一样的地方就是，osc就是光标位置变化请求一次帧。pot和vlc的滑块并不是一移动就请求一次帧

**Reply by Anikatball** [reply]
> pot启动速度就是慢一大截，跟配置关联性不大。
> 
> mpv默认状态下，硬盘速度也会影响seek，放固态盘比机械盘快, 内存够大的话, 倒是有个可提升seek速度的点， 给大点缓存，启动后全加载到内存里，seek的时候就少了读盘的步骤，速度应该会有所提升。
> 
> ```ini
> cache=yes   # 开启缓存，默认值auto只会在播放网络链接时开启
> demuxer-max-bytes=4096MiB       # 向后缓存大小 
> demuxer-max-back-bytes=1024MiB  # 前面保留的缓存大小
> ```

感谢反馈，看来不是我的配置问题，pot启动真的是太慢了，我试过安装旧版pot启动能稍微快一点但还是远比不上mpv.......

硬盘速度影响seek我也发现了，我通过SMB共享从机械硬盘里读取的时候，进度条会有条状块在移动，这是在缓存吧。我试试加大缓存！

**Reply by verygoodlee** [reply]
是的，OSC的进度条一般都会显示缓存的状态，或者可以在STATS第3页查看更详细的缓存区间
![屏幕截图 2024-09-03 001426](https://github.com/user-attachments/assets/ec0a6ab0-3012-4933-b76b-182a4325e8d8)

**Comment by hooke007** [comment]
我白天用i3试了试，尽可能近似设置下没感觉出区别，相反mpv是响应最快的那个。。。

**Comment by Trudeau566** [comment]
hr-seek = no

**Reply by hooke007** [reply]
这与uosc界面的操控无关
